src/sentry/api/endpoints/organization_tags.py

```diff
@@ -27,6 +32,9 @@ def get(self, request: Request, organization) -> Response:
         except NoProjects:
             return Response([])
 
+        if request.GET.get("dataset") and request.GET.get("dataset") not in Dataset:
+            raise InvalidDatasetParam()
```

Comment:
I think it would be better to raise a `ParseError` here instead, which sets the response as a 400 (and you'll want to include a message as well)